torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -901,25 +948,44 @@ class TritonHOPifier:
     TritonHOPifier is an abstract class that can be overriden by its subclasses.
     """
 
-    def raise_unsupported(self, msg):
+    def raise_unsupported(self, msg: str) -> None:
         raise NotImplementedError("abstract method")
 
-    def is_callable(self, maybe_callable):
+    def is_callable(self, maybe_callable: Any) -> bool:
         raise NotImplementedError("abstract method")
 
-    def get_value(self, val):
+    def get_value(self, val: Any) -> Any:
         raise NotImplementedError("abstract method")
 
-    def call_grid(self, grid, meta, tx):
+    def call_grid(
+        self,
+        grid: "Union[TritonGridCallableType, VariableTracker]",
+        meta: "Union[TritonMetaParamsType, VariableTracker]",
```

Comment:
is there a way to specify... if one of these is a variable tracker then the other should be too?

Reply:
Yes:
```
@overload
def call_grid(self, grid: TritonGridCallableType, meta: TritonMetaParamsType, tx: Optional["InstructionTranslator"]) -> Union[Tuple[Union[int, sympy.Expr, SymInt], ...], Tuple["Proxy", ...]]:
    ...
@overload
def call_grid(self, grid: VariableTracker, meta: VariableTracker, tx: Optional["InstructionTranslator"]) -> Union[Tuple[Union[int, sympy.Expr, SymInt], ...], Tuple["Proxy", ...]]:
    ...
def call_grid(
        self,
        grid: Union[TritonGridCallableType, VariableTracker],
        meta: Union[TritonMetaParamsType, VariableTracker],
        tx: Optional["InstructionTranslator"],
    ) -> Union[Tuple[Union[int, sympy.Expr, SymInt], ...], Tuple["Proxy", ...]]:
```
When you use `overload` the type-checker picks one of the overloaded APIs and ignores the non-overloaded one.  At runtime python ignores the overloaded APIs and uses the non-overloaded one.

Reply:
I just marked this `type: ignore[no-untyped-def]` - see my comment in response to Richard